File: app/src/main/java/cn/surine/schedulex/ui/schedule_init/ScheduleInitViewModel.java
package cn.surine.schedulex.ui.schedule_init;

import android.text.Editable;
import android.text.TextWatcher;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import cn.surine.schedulex.base.utils.SimpleTextWatcher;

/**
 * Intro：
 *
 * @author sunliwei
 * @date 2020-01-22 19:09
 */
public class ScheduleInitViewModel extends ViewModel {
    public MutableLiveData<String> scheduleName = new MutableLiveData<>();

    public TextWatcher scheduleWatcher = new SimpleTextWatcher() {
        @Override
        public void afterTextChanged(Editable s) {
            scheduleName.setValue(s.toString());
        }
    };

}

File: app/src/main/java/cn/surine/schedulex/ui/schedule_init/ScheduleInitViewModel.kt
package cn.surine.schedulex.ui.schedule_init

import android.text.Editable
import androidx.lifecycle.MutableLiveData
import cn.surine.schedulex.base.controller.BaseViewModel
import cn.surine.schedulex.base.utils.SimpleTextWatcher

/**
 * Intro：
 *
 * @author sunliwei
 * @date 2020/6/24 16:01
 */
class ScheduleInitViewModel :BaseViewModel(){
    val scheduleName = MutableLiveData<String>()
    val scheduleWatcher = object : SimpleTextWatcher(){
        override fun afterTextChanged(s: Editable?) {
            super.afterTextChanged(s)
            scheduleName.value = s.toString()
        }
    }
}